default/python/AI/freeorion_tools/_freeorion_tools.py

```diff
@@ -143,10 +144,12 @@ def chat_human(message):
     debug("Chat Message to human: %s", remove_tags(message))
 
 
-def cache_by_session(func):
+def cache_for_session(func):
```

Comment:
Now when we got python3 we can just:
`cache_for_session = functools.lru_cache`

https://docs.python.org/3/library/functools.html?highlight=lru#functools.lru_cache

Reply:
Beyond the scope of this PR but probably good to keep in mind for the future.